components/script/dom/formdata.rs

```diff
@@ -163,4 +164,35 @@ impl FormData {
 
         ret
     }
+
+    pub fn atoms(&self) -> Vec<Atom> {
+        let mut ret = vec![];
+        for (key, value) in self.data.borrow().iter() {
```

Comment:
Oh I wrote it in this way mainly because the `append` requires a mutable borrow as input, not so good style ... But maybe `flat_map` or something can work.